### ISSUE TITLE
SQL Server 2017: Create database outside transaction

### ISSUE THREAD

**Issue by lunderhage** [lecaillon/Evolve#186]
I am trying to create a database in SQL Server 2017 after have reading this PR:

https://github.com/lecaillon/Evolve/pull/150/files

```
-- evolve-tx-off
CREATE DATABASE people_registry;

```
Database is created, but:
```
Executing Migrate...
Evolve initialized.
No metadata found.
Successfully applied migration V1__create_db.sql in 1093 ms.
Cannot release the application lock (Database Principal: 'public', Resource: 'Evolve') because it is not currently held. Sql query: sp_releaseapplock Cannot release the application lock (Database Principal: 'public', Resource: 'Evolve') because it is not currently held.
Cannot release the application lock (Database Principal: 'public', Resource: 'Evolve') because it is not currently held. Sql query: sp_releaseapplock Cannot release the application lock (Database Principal: 'public', Resource: 'Evolve') because it is not currently held.
```

Why does Evolve think it has the lock here and why does it fail because it did not? I appreciate some guidance here.

**Comment by lecaillon** [comment]
Hi,
I'm surprised by this behavior because I have a test [here ](Run_all_SQLServer_migrations_work)that uses this [migration](https://github.com/lecaillon/Evolve/blob/master/test/Evolve.Tests/Integration/SQLServer/Resources/Sql_Scripts/Migration/V3_1__create_database_evolvedb.sql).
``` sql
-- evolve-tx-off
IF(db_id(N'evolvedb') IS NOT NULL)
BEGIN
	DROP DATABASE evolvedb;
END;

CREATE DATABASE evolvedb;
```
Is this the only statement you have in the migration file ?
Maybe you can give me more insight about the context of your migration : os, database etc ?

**Comment by lunderhage** [comment]
Yes, that is the only sql script I have in Sql_Scripts. I am just trying to create a new database from scratch. I want to make a demo for my colleague with some different databases. CockroachDB was dead easy.

SQL Server (Linux) is:
```
version: "3.4"
services:
  sql-server:
    image: mcr.microsoft.com/mssql/server:2017-latest
    environment:
      - ACCEPT_EULA=Y
      - SA_PASSWORD=1T2e3S4t
    ports:
      - 1433:1433
    volumes:
      - sql-server-data:/var/opt/mssql
volumes:
  sql-server-data:
```

**Comment by lecaillon** [comment]
Weird, I even try to take the tag **2017-latest** you use to run my test. And it works.

Does the tag `-- evolve-tx-off` located in the first line of your script ?

**Comment by lecaillon** [comment]
I think your issue is caused by the release of the lock placed at application level when Evolve starts, unitil it ends. That part is located here in the [code](
https://github.com/lecaillon/Evolve/blob/7ddd68d79303511d5857027602a196f5c6481cfd/src/Evolve/Dialect/SQLServer/SQLServerDatabase.cs#L26). 
It is like Evolve can get the lock at start, but this one is already released after you created the database, which caused the failure of the stored procedure `sp_releaseapplock`
As I cannot reproduce your issue, it would be great if you can : 
- call `sp_getapplock` with the paramters Evolve uses
- try to get the same lock again to check you cannot obviously
- execute your statement (CREATE DATABASE...)
- check you have the lock again. If not, it means the lock is released automatically by a CREATE DATABASE statement. _(But not in my tests ...?)_

**Comment by lunderhage** [comment]
Something funny is going on in my database server here:

This works fine:
```
DROP DATABASE IF EXISTS Trams;

sp_getapplock @Resource="Evolve", @LockOwner=Session, @LockMode=Exclusive, @LockTimeout=0;

CREATE DATABASE Trams;

sp_releaseapplock @Resource="Evolve", @LockOwner=Session****
```

But if I do
```
sp_getapplock @Resource="Evolve", @LockOwner=Session, @LockMode=Exclusive, @LockTimeout=0;
```
in another session and the run the above script it will fail with 
```
SQL Error [1223] [S0001]: Cannot release the application lock (Database Principal: 'public', Resource: 'Evolve') because it is not currently held.
```
I know that I get the lock (0) by looking at @@Error.

**Comment by lunderhage** [comment]
It would be really useful to get the actual SQL being executed.

**Comment by lecaillon** [comment]
The lock used is a lock by session. If the session changes, it can explain the error. But again, how is it possible that this [test ](https://github.com/lecaillon/Evolve/blob/master/test/Evolve.Tests/Integration/SQLServer/MigrationTest.cs) works ? For what I understand, it is really similar to what you try to achieve ? Check the that [migration script](https://github.com/lecaillon/Evolve/blob/master/test/Evolve.Tests/Integration/SQLServer/Resources/Sql_Scripts/Migration/V3_1__create_database_evolvedb.sql).

The lock is used to coordinate the migrations on multiple nodes. This prevents 2 distinct Evolve executions from executing a migration on the same database at the same time. So if you do not need this feature, you can deactivate it, by setting [Evolve.EnableClusterMode](https://evolve-db.netlify.app/configuration/) to false. It will fix your issue.

**Comment by lecaillon** [comment]
Addressed in #97
